llvm/include/llvm/CodeGen/GlobalISel/CallLowering.h

```diff
@@ -407,6 +407,12 @@ class CallLowering {
       CallingConv::ID CallConv, bool IsVarArg,
       ArrayRef<Register> ThisReturnRegs = std::nullopt) const;
 
+  bool determineAndHandleAssignments(
```

Comment:
Now that I think of it, I think it would be better to just use separate determineAssignments, and handleAssignments. determineAndHandleAssignments is already the simple case where the CCState doesn't need any custom handling

Reply:
Still should undo this and just use the 2 separate pieces

Reply:
Done. See commit [df3a52e](https://github.com/llvm/llvm-project/pull/72461/commits/df3a52e8f182b85b80d7037b2b68c394de267fd9)